lib/pacemaker/pcmk_sched_group.c

```diff
@@ -478,43 +478,69 @@ pcmk__group_apply_coloc_score(pe_resource_t *dependent,
     }
 }
 
+/*!
+ * \internal
+ * \brief Return action flags for a given group resource action
+ *
+ * \param[in,out] action  Group action to get flags for
+ * \param[in]     node    If not NULL, limit effects to this node
+ *
+ * \return Flags appropriate to \p action on \p node
+ */
 enum pe_action_flags
-group_action_flags(pe_action_t *action, const pe_node_t *node)
+pcmk__group_action_flags(pe_action_t *action, const pe_node_t *node)
 {
-    GList *gIter = NULL;
-    enum pe_action_flags flags = (pe_action_optional | pe_action_runnable | pe_action_pseudo);
+    // Default flags for a group action
+    enum pe_action_flags flags = pe_action_optional
```

Comment:
It would be kind of nice to set this up in such a way that we're not managing two sets of flags here (`flags` (return value) and `action->flags`)... there's probably a way, but it might turn out to be less clear and it's definitely not necessary for this PR.

Reply:
I think the main complication is that for clone and bundle actions, the action flags can vary by node, so action->flags (independent of node) and the return value of ->action_flags() (considering node) need to be tracked separately. That could be commented better for sure.

An alternative design might be for action->flags to be a hash table by node, but I haven't looked into how feasible that might be.

Reply:
> That could be commented better for sure.

It's somewhat commented in `libpacemaker_private.h` above the method declaration. The current design probably works fine but at a glance it looks open to weirdness.

For example, note that we initialize `flags` to `pe_action_optional|pe_action_runnable|pe_action_pseudo`. If it is (or were) possible for `action->flags` to have the optional bit **un**-set, while the optional bit is set for all the member resources, then we would never unset it from `flags`. Then our return value contains `optional` even though `action->flags` is marked as mandatory. Similar story with the runnable bit.

It might not be possible for this to happen in practice, but it makes the function a little harder to reason about.